x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/rules_table_context.tsx

```diff
@@ -36,12 +38,17 @@ import {
 } from './rules_table_defaults';
 import { RuleSource } from './rules_table_saved_state';
 import { useRulesTableSavedState } from './use_rules_table_saved_state';
+import * as i18n from './translations';
+
+export interface ExtendedRule extends Rule {
+  snoozeSettings?: RuleSnoozeSettings | Error;
+}
```

Comment:
Let's avoid using inheritance unless there was a specific reason for doing that.

If you need to colocate data, you could do it like that:

```
interface RuleData {
  rule: Rule;
  snoozeSettings: RuleSnoozeSettings | undefined;
}
```

However, I'm not sure why we'd need to do that. I think keeping a separate list of snooze settings + all the additional react-query state for it (`isFetching`, `isLoading`, etc) would be a simpler solution. On read, the `RuleSnoozeBadge` component would just need to look up the snooze settings in a map by a rule id. This should be super fast and shouldn't require any memorization (like the one you do in `useFindExtendedRules`).

Also, please note that from the `RulesTableState` we expose properties like `isFetched`, `isFetching`, `isLoading`, `isRefetching`. Not fully sure about that, but I think these might need to be `true` when either of the loading processes (rules or snooze settings) is ongoing.